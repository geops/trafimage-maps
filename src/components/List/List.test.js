import React from 'react';
import { render } from '@testing-library/react';
import List from './List';

describe('List', () => {
  describe('when no properties are set', () => {
    let spy = null;

    beforeEach(() => {
      window.console.error = jest.fn().mockImplementation(() => {});
      spy = jest.spyOn(window.console, 'error');
    });

    afterEach(() => {
      spy.mockRestore();
      window.console.error.mockRestore();
    });

    test('displays 1 error for required property ', () => {
      render(<List />);
      expect(spy).toHaveBeenCalledTimes(1);
    });

    test('matches snapshot', () => {
      const component = render(<List />);
      expect(component.container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when properties are set', () => {
    const defaultItems = [
      {
        label: 'foo',
      },
      {
        label: 'bar',
      },
      {
        label: 'foo2',
      },
    ];

    const items = [
      {
        label: 'qux',
      },
      {
        label: 'quux',
      },
      {
        label: 'corge',
      },
    ];

    test('matches snapshot with defaultItems', () => {
      const component = render(
        <List
          className="tm-foo"
          defaultItems={defaultItems}
          renderTitle={() => 'my_foo_title'}
          renderItem={(item) => item.label}
          getItemKey={() => Math.random()}
          onSelect={() => {}}
        />,
      );
      expect(component.container.innerHTML).toMatchSnapshot();
    });

    test('matches snapshot with items', () => {
      const component = render(
        <List
          className="tm-foo"
          defaultItems={defaultItems}
          items={items}
          renderTitle={() => 'my_foo_title'}
          renderItem={(item) => item.label}
          getItemKey={() => Math.random()}
          onSelect={() => {}}
        />,
      );
      expect(component.container.innerHTML).toMatchSnapshot();
    });
  });
});
