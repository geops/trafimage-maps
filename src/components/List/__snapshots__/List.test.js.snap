// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List when no properties are set matches snapshot 1`] = `""`;

exports[`List when properties are set matches snapshot with defaultItems 1`] = `""`;

exports[`List when properties are set matches snapshot with items 1`] = `
<ul role="menu"
    class="tm-foo"
>
  <li class="tm-list-item"
      role="menuitem"
      tabindex="0"
  >
    qux
  </li>
  <li class="tm-list-item"
      role="menuitem"
      tabindex="0"
  >
    quux
  </li>
  <li class="tm-list-item"
      role="menuitem"
      tabindex="0"
  >
    corge
  </li>
</ul>
`;
